lib/package.rb

```diff
@@ -183,55 +185,51 @@ def self.depends_on (dependency = nil)
     @dependencies
   end
 
-  def self.get_url (architecture)
+  def self.get_url(architecture)
     if !@build_from_source and @binary_url and @binary_url.has_key?(architecture)
-      return @binary_url[architecture]
+      @binary_url[architecture]
+    elsif @source_url.respond_to?(:has_key?)
+      @source_url.has_key?(architecture) ? @source_url[architecture] : nil
     else
-      if @source_url.respond_to?(:has_key?)
-        return @source_url.has_key?(architecture) ? @source_url[architecture] : nil
-      else
-        return @source_url
-      end
+      @source_url
     end
   end
 
-  def self.get_binary_url (architecture)
-    return @binary_url.has_key?(architecture) ? @binary_url[architecture] : nil
+  def self.get_binary_url(architecture)
+    @binary_url.has_key?(architecture) ? @binary_url[architecture] : nil
   end
 
-  def self.get_source_url (architecture)
-    return @source_url.has_key?(architecture) ? @source_url[architecture] : nil
+  def self.get_source_url(architecture)
+    @source_url.has_key?(architecture) ? @source_url[architecture] : nil
   end
 
-  def self.get_sha256 (architecture)
+  def self.get_sha256(architecture)
     if !@build_from_source and @binary_sha256 and @binary_sha256.has_key?(architecture)
-      return @binary_sha256[architecture]
+      @binary_sha256[architecture]
+    elsif @source_sha256.respond_to?(:has_key?)
+      @source_sha256.has_key?(architecture) ? @source_sha256[architecture] : nil
     else
-      if @source_sha256.respond_to?(:has_key?)
-        return @source_sha256.has_key?(architecture) ? @source_sha256[architecture] : nil
-      else
-        return @source_sha256
-      end
+      @source_sha256
     end
   end
 
   def self.get_extract_dir
-    name + '.' + Time.now.utc.strftime("%Y%m%d%H%M%S") + '.dir'
+    name + '.' + Time.now.utc.strftime('%Y%m%d%H%M%S') + '.dir'
   end
 
-  def self.is_binary? (architecture)
+  def self.is_binary?(architecture)
     if !@build_from_source and @binary_url and @binary_url.has_key?(architecture)
```

Comment:
This can be simplify to `return ( !@build_from_source and @binary_url and @binary_url.has_key?(architecture) )`